### PR TITLE
Fix: don't set upper bounds

### PR DESCRIPTION
**Pull request by agoose77**
Upper bounds are problematic in the Python ecosystem: https://iscinumpy.dev/post/bound-version-constraints/

Also, I've removed `ipykernel` as a dependency, and replaced it with `IPython`

### CODE REVIEW

**Review by kolibril13 — APPROVED** [review]
Thanks for this contribution!

**Comment by kolibril13** [comment]
I've just seen that the test does fail:
```console
SolverProblemError

  The current project's Python requirement (>=3.8) is not compatible with some of the required packages Python requirement:
    - isort requires Python >=3.6.1,<4.0, so it will not be satisfied for Python >=4.0
  
  Because no versions of isort match >5.10.1
   and isort (5.10.1) requires Python >=3.6.1,<4.0, isort is forbidden.
  So, because jupyter-splitview depends on isort (>=5.10.1), version solving failed.
```
@agoose77 Any ideas what to do about this?
I want to stay with poetry, so is there a better way to define versions without getting this `SolverProblemError` ?

File: pyproject.toml [hunk 1]
```diff
@@ -6,17 +6,17 @@ authors = ["kolibril13 <44469195+kolibril13@users.noreply.github.com>"]
 license = "MIT"
 
 [tool.poetry.dependencies]
-python = "^3.8,<3.11"
-ipykernel = "^6.13.0"
-jupyterlab = "^3.3.4"
+python = ">=3.8"
+jupyterlab = ">=3.3.0"
```
Comment (kolibril13) [inline pyproject.toml:10]:
```suggestion
```
Can the JupyterLab dependency be removed?

**Comment by agoose77** [comment]
> I've just seen that the test does fail:
> 
> ```
> SolverProblemError
> 
>   The current project's Python requirement (>=3.8) is not compatible with some of the required packages Python requirement:
>     - isort requires Python >=3.6.1,<4.0, so it will not be satisfied for Python >=4.0
>   
>   Because no versions of isort match >5.10.1
>    and isort (5.10.1) requires Python >=3.6.1,<4.0, isort is forbidden.
>   So, because jupyter-splitview depends on isort (>=5.10.1), version solving failed.
> ```
> 
> @agoose77 Any ideas what to do about this? I want to stay with poetry, so is there a better way to define versions without getting this `SolverProblemError` ?

This is an annoying problem in the Python packaging ecosystem right now, mainly with Poetry and `pdm`. The explanation can be found [here](https://discuss.python.org/t/requires-python-upper-limits/12663/72), but the short summary is that the Python version constraint is designed only to keep older Python's working when new incompatible packages are released. This means it should only be a lower bound. Unfortunately, Poetry expects you to set an upper bound in the case that any dependencies set an upper bound, which means you'll ultimately have to set it for now.

File: pyproject.toml [hunk 2]
```diff
@@ -6,17 +6,16 @@ authors = ["kolibril13 <44469195+kolibril13@users.noreply.github.com>"]
 license = "MIT"
 
 [tool.poetry.dependencies]
-python = "^3.8,<3.11"
-ipykernel = "^6.13.0"
-jupyterlab = "^3.3.4"
+python = ">=3.8"
```
Comment (agoose77) [inline pyproject.toml:9]:
```suggestion
python = "^3.8"
```

Comment (agoose77) [reply pyproject.toml:9]: Fix for Poetry

File: pyproject.toml [hunk 3]
```diff
@@ -6,17 +6,16 @@ authors = ["kolibril13 <44469195+kolibril13@users.noreply.github.com>"]
 license = "MIT"
 
 [tool.poetry.dependencies]
-python = "^3.8,<3.11"
-ipykernel = "^6.13.0"
-jupyterlab = "^3.3.4"
+python = "^3.8"
```
Comment (agoose77) [inline pyproject.toml:9]: And again, scipy also sets the upper bound, so we need to further constrain as you did originally.

```suggestion
python = ">=3.8,<3.11"
```

**Comment by kolibril13** [comment]
Test is passing, I think this can be merged now?

**Pull request merged**